### PR TITLE
Makefile: make 'make check' faster

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Try to accelerate the unit test. 
`make check` is painfully slow, I make it faster in this commit. 

Before:

```
➜  tidb git:(master) ✗ time make check    
make check  44.87s user 34.64s system 86% cpu 1:31.63 total
```

After:
```
➜  tidb git:(make-check-accelerate) ✗ time make check
make check  6.87s user 1.77s system 113% cpu 7.607 total
```

### What is changed and how it works?

* remove vet check
* make lint check on ./...

#### 1.  vet is removed from `make check`, because from [Go1.10](https://golang.org/doc/go1.10#test) on, `vet` will run automatically when `go test` is called, and the [Go1.11](https://golang.org/doc/go1.11#test) also keeps this behavior.

To verify that, I tried this:

```
➜  session git:(master) ✗ cd session
➜  session git:(master) ✗ time go test -vet=off 1>/dev/null 2>/dev/null
go test -vet=off > /dev/null 2> /dev/null  7.26s user 1.28s system 63% cpu 13.403 total

➜  session git:(master) ✗ time go test 1>/dev/null 2>/dev/null   
go test > /dev/null 2> /dev/null  12.54s user 1.80s system 91% cpu 15.719 total
```

You can see that test with `-vet=off` is much faster.
Since each test package has already run `vet`, let `make check` run it is meaningless 

#### 2. `make lint` run this command:

> 	@CGO_ENABLED=0 retool do revive -formatter friendly -config revive.toml $(PACKAGES)

I find it downloads $(PACKAGES) from network, which is not a desired behavior.

To verify that, I tried to lint a non-local package (github.com/tiancaiamao/shen-go/runtime for example):

```
➜  tidb git:(master) ✗ retool do revive -formatter friendly -config revive.toml github.com/tiancaiamao/shen-go/runtime
go/build: importGo github.com/tiancaiamao/shen-go/runtime: exit status 1
go: finding github.com/tiancaiamao/shen-go/runtime latest
go: finding github.com/tiancaiamao/shen-go latest
go: extracting github.com/tiancaiamao/shen-go v0.0.0-20181124091523-b5e263ee27d0
-> unzip /media/genius/OS/project/pkg/mod/cache/download/github.com/tiancaiamao/shen-go/@v/v0.0.0-20181124091523-b5e263ee27d0.zip: malformed file path "ShenOSKernel-21.0/tests/depth'.shen": invalid char '\''
can't load package: package github.com/tiancaiamao/shen-go/runtime: unknown import path "github.com/tiancaiamao/shen-go/runtime": cannot find module providing package github.com/tiancaiamao/shen-go/runtime
```

#### 3. Another change is that I call `revive` directly instead of under the `retool do`.
Retool is introduced in https://github.com/pingcap/tidb/pull/7240, to specify tool version, now that we have Go module, it's possible to specify tool version in go.mod

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
 - No code

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8755)

### CODE REVIEW

**Comment by tiancaiamao** [comment]
@jackysp @lysu @gregwebs

**Comment by lysu** [comment]
@tiancaiamao please fix CI failure

**Comment by gregwebs** [comment]
I just tested this out. `make check` seems consistently a little bit slower for me than running directly. Running `revive` directly or via retool, but just not via the Makefile is equally fast by a little bit for me. Is the third change necessary? If so, we really need to understand why and find a better solution for managing our tooling.

```
make lint                                                                      18.78s user 0.69s system 516% cpu 3.766 total
revive -formatter friendly -config revive.toml                                 17.16s user 0.52s system 653% cpu 2.705 total
retool do revive -formatter friendly -config revive.toml                       17.80s user 0.46s system 659% cpu 2.769 total
```

Whenever possible in other projects I try to avoid creating a special environment inside a Makefile that only works in the Makefile. I don't know why the Makefile version is consistently slower.

File: cmd/explaintest/main.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ import (
 
 	"flag"
 	"github.com/go-sql-driver/mysql"
+	_ "github.com/mgechev/revive/lint"
```
[the remark this answers is not in the document]
Comment (tiancaiamao) [reply cmd/explaintest/main.go]: Done

**Comment by tiancaiamao** [comment]
@gregwebs 
I thought `make lint` on $(PACKAGES) is slow because it download packages from the network, but it seems I'm biased. 
And I'm not quite sure use  `./...` works.  Now I change `make lint` to work on $(FILES). 
Anyway, check $(PACKAGES) is quite slow, check $(FILES) is much better.

Before, on $(PACKAGES):
```
➜  tidb git:(make-check-accelerate) ✗ time make lint     
linting
make lint  25.51s user 28.43s system 74% cpu 1:12.65 total
```
After, on $(FILES)
```
➜  tidb git:(make-check-accelerate) ✗ time make lint
linting
make lint  5.00s user 1.02s system 166% cpu 3.609 total
```

After the change, `make lint` reports some errors for me:
```
linting
  ✘  https://revive.run/r#var-naming  var errTestSql should be errTestSQL  
  executor/set_test.go:53:2

  ✘  https://revive.run/r#var-naming  don't use underscores in Go names; var info_tables should be infoTables  
  infoschema/infoschema_test.go:250:2

  ✘  https://revive.run/r#var-naming  var stmtId should be stmtID  
  executor/prepared_test.go:87:3

  ✘  https://revive.run/r#superfluous-else  if block ends with a break statement, so drop this else and outdent its block  
  planner/core/prepare_test.go:262:10

  ✘  https://revive.run/r#var-naming  var globalAutoId should be globalAutoID  
  meta/autoid/autoid_test.go:63:2

  ✘  https://revive.run/r#var-naming  var globalAutoId should be globalAutoID  
  meta/autoid/autoid_test.go:154:2

  ✘  https://revive.run/r#var-naming  don't use underscores in Go names; var err_admin should be errAdmin  
  executor/executor_test.go:242:5

  ✘  https://revive.run/r#var-naming  don't use underscores in Go names; var create_time should be createTime  
  executor/show_test.go:624:2

  ✘  https://revive.run/r#var-naming  don't use underscores in Go names; var create_time should be createTime  
  executor/show_test.go:666:2

  ✘  https://revive.run/r#var-naming  var explainedSql should be explainedSQL  
  executor/merge_join_test.go:218:2

  ✘  https://revive.run/r#var-naming  var explainedSql should be explainedSQL  
  executor/merge_join_test.go:228:2

  ✘  https://revive.run/r#var-naming  don't use underscores in Go names; type mockPump_PullBinlogsClient should be mockPumpPullBinlogsClient  
  session/session_test.go:103:6

✘ 12 problems (12 errors, 0 warnings)
```

That sounds good.

-----------------

As for the retool, you know, the CI runs on a new clean environment every time, and it downloads files from the network.
It takes @iamxy a lot of effort to cache Go module in our own CI environment to skip the downloading.
If we put revive package under go.mod, we can benefit from @iamxy 's great work.

**Comment by gregwebs** [comment]
Tool usage workflows are unfortunately complex in go. We should continue the discussion about whether the retool approach is best. But this should be a separate PR or discussion.

revive should use the same system as the rest of the linting tools. So we should change how we use all of the tools or in a separate PR and keep revive under retool in this PR.


The current PR removes tool isolation in two ways 1) global install 2) potential conflict between tool dependencies with binary dependencies.
If we look at this problem a little more closely as a separate issue instead of rushing code through in this PR, I think we can come up with a better way.

**Comment by tiancaiamao** [comment]
I'm no discussing about the retool approach in this PR.
The purpose of this PR is clear: make 'make check' faster.

> The current PR removes tool isolation in two ways 
> 1) global install 
> 2) potential conflict between tool dependencies with binary dependencies.

No "global install" problem, because CI runs in a totally new clean environment, and the environment is destroyed after that. 
Maybe if I could put the revive binary in other place and it's more acceptable? 

I'll make a separate go.mod for tools binary, to address the dependencies pollution problem.
I promise to do it in the next PR, is it ok?
@gregwebs

**Pull request merged**